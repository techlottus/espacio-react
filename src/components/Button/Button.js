

import React, { useEffect } from 'react'


export const Button = React.memo(({ data, onClick }) => {



  const buttonRef = React.createRef();
  
  useEffect(() => {
    buttonRef.current.data = {
      id: data.id || null,
      type: data.type  || 'primary',
      title: data.title || '',
      size: data.size || 'small',
      icon: data.icon || '',
      lyIcon: data.lyIcon,
      disabled: data.disabled,
      isExpand: data.isExpand,
      test: data.test || ''
    }
  }, [data]) // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    buttonRef.current.addEventListener('onClick', onClick);
  }, []) // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-button ref={buttonRef}></lottus-button>
    </>
  )
})


