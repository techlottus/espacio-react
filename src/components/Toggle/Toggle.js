import React, { useEffect } from 'react';



export const Toggle = React.memo(({data, onSwitch}) => {

    const toggleRef = React.createRef();
  
    useEffect(() => {
      toggleRef.current.data = {
        size: data.size ||'',
        status: data.status,
        disabled: data.disabled,
        test: data.test || '',
      };
    },[data]) // eslint-disable-line react-hooks/exhaustive-deps 

    useEffect(() => {
      toggleRef.current.addEventListener('onSwitch', onSwitch)
    },[]) // eslint-disable-line react-hooks/exhaustive-deps
  
    return (
      <>
        <lottus-toogle-switch ref={toggleRef}></lottus-toogle-switch>
      </>
    )
  });