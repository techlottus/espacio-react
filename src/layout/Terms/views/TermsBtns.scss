@import '../../../App.scss';

.terms-container {
  padding: pixeltorem(48px) pixeltorem(5px);
  margin: 0 pixeltorem(15px);
  font-family: "Nunito", sans-serif;

  @media only screen and (min-width: map-get($grid-breakpoints, m)) {
    padding: pixeltorem(48px) pixeltorem(5px) 0 pixeltorem(5px);

  }

  .termsbtns-title{
    font-family: 'Bebas',sans-serif;
    font-size: pixeltorem(42px);
    padding-bottom: pixeltorem(5px);

    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
          font-size: pixeltorem(56px);
      }
    
  }
  .termsbtns-text {
    font-size: pixeltorem(14px);
    line-height: normal;
  }

  .buttons {
    padding: pixeltorem(15px) 0 pixeltorem(10px) 0;

    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
      padding: pixeltorem(20px) 0;
    }

    .button-wrap {
        padding: pixeltorem(8px) 0;

        button{
          width: 100%;
          text-align: center;
          justify-content: center;

          @media only screen and (min-width: map-get($grid-breakpoints, m)) {
            width: auto;
           }
        }
    }
  }
  .newsletter-wrap{
    display: flex;
    align-items: center;

    > .termsbtns-text{
      padding-left: pixeltorem(14px);
    }
  }

  .progress-bar{
    padding: pixeltorem(36px) 0 pixeltorem(15px) 0;

    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
      padding: pixeltorem(35px) 0;
    }
  }

  .send-btn{
      button{
        width: 100%;
        text-align: center;
        justify-content: center;

        @media only screen and (min-width: map-get($grid-breakpoints, m)) {
         width: auto;
        }
      }
  }
}
