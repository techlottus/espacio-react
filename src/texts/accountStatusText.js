export const accountStatusText = {
  statusTitle: "Estado de cuenta y consultas de facturas",
  resumeTitle: "Resumen de adeudo",
  total: "Total",
  errorMessage:
    "Tuvimos dificultades para cargar tu información en este momento",
  emptyMessage: {
    congrats: "Felicidades, estás al día con tus pagos",
    continue: "Continúa transformando tu futuro",
  },
  images: { 
    user: "images/user.png",
    empty: "yeah.png",
  },
  btnModal: {
    delete: "Eliminar",
    cancel: "Cancelar",
  },
  btnDownload: {
    download: "Descargar Estado de cuenta",
  },
  btnInvoice: {
    checkInvoice: "Consulta tus facturas",
  },
  tableAccountStatus: [
    { title: "Clave" },
    { title: "Ciclo" },
    { title: "Periodo" },
    { title: "Adeudo" },
  ],

  periodDetailTitle: "Detalles por periodo",
  periodDetailDescription:
    "Selecciona el programa y el periodo para ver los detalles de los movimientos; se clasifican de la siguiente manera:",
  periodDetailBullets: {
    debt: "Con adeudo",
    payed: "Pagados",
    toPay: "Por pagar",
  },
  periodDetailSelects: {
    program: "Programa",
    period: "Periodo",
  },
  accountDetailPeriod: {
    title: "Periodo",
    icon: "calendar_today",
    debt: "Saldo del periodo",
  },
  accountDetailMovements: {
    title: "Movimientos",
  },
  tableInvoiceQuery: [
    { title: "Fecha de emisión" },
    { title: "Periodo" },
    { title: "Concepto" },
    { title: "Monto" },
    { title: "RFC" },
    { title: "Preview" },
    { title: "PDF" },
    { title: "XML" },
  ],
  tableRowInvoiceQuery: [
    { title: "07 Abril 2021" },
    { title: "Periodo 5" },
    { title: "Colegiatura" },
    { title: "$2405.50" },
    { title: "LRKO485767HF2" },
  ],
  accountStatusInvoiceQuery: {
    title: "Consulta de facturas",
    description:
      "Filtra por rango de fechas, selecciona el perido que deseas consultar y presiona descargar.",
  },
  btnInvoiceQuery: {
    download: "Descargar todo",
    clear: "Limpiar",
  },
  linkInvoiceQuery: "Ver más"
};
