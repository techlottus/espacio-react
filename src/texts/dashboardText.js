export const dashboardText = {
  menu: {
    aula: {
      emoji: "mortar-board",
      text: "Aula",
    },
    book: {
      emoji: "books",
      text: "Biblioteca",
    },
    helpCenter: {
      emoji: "thought-balloon",
      text: "Centro de Ayuda",
    },
    procedures: {
      emoji: "file-folder",
      text: "Trámites",
    },
    accountStatus: {
      emoji: "chart-with-upwards-trend",
      text: "Estado de Cuenta",
    },
    payment:  {
      emoji: "credit-card",
      text: "Pagos",
    },
    invoice: {
      emoji: "page-facing-up",
      text: "Facturas"
    },
    benefits: {
      emoji: "star",
      text: "Portal de Beneficios"
    },
    sede: {
      emoji: "notebook",
      text: "SeDe"
    },
    psychopedagogical: {
      emoji: "blue-heart",
      text: "Centro de Atención Psicopedagógica"
    },
    evaluateServices: {
      emoji: "ec ec-white-check-mark",
      text: "Evalúa los servicios"
    },
    registrationSubjects: {
      emoji: "ec ec-closed-book",
      text: "Inscribe tus materias"
    },
    logout: {
      emoji: "back",
      text: "Cerrar sesión",
    },
  },
  titleWelcome: "Hola",
  images: {
    cardHeader: "images/header.png",
    empty: "empty.png",
  },
  logo: {
    urlLogo: "logo.svg"
  },
  logos: "logo.svg",
  sectionDashboard: {
    titleNews: "Noticias de tu universidad",
    titleCards: "¿Qué quieres hacer hoy?",
  },
  playIcon: "play_circle_outline",
  cards: {
    one: {
      title: "Aula",
      icon: "school",
    },
    two: {
      title: "Pagos",
      icon: "credit_card",
    },
    three: {
      title: "Biblioteca",
      icon: "import_contacts",
    },
    four: {
      title: "Trámites",
      icon: "post_add",
    },
    five: {
      title: "Inscribe tus materias",
      icon: "add_circle_outline",
    },
  },
  footerText: {
    text: "©2022 Derechos reservados Lottus Education",
    privacy: "Aviso de privacidad y términos y condiciones",
  },
};