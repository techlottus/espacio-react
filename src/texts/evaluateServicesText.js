export const evaluateServicesText = {
    evaluateServicesTitle: "Evalua tus servicios digitales",
    modalSurvey: {
        title: "Evalua tu experiencia para realizar tu pago a través de Espacio UTC",
        text: "¿Que tan probable es que recomiendes los pagos en Espacio UTC?"
    }
}