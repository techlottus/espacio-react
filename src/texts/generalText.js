export const generalText = {
  modals: {
    modalExit: {
      icon: "help",
      modalTitlePayment: "¿Seguro que deseas salir de la sección de pagos?",
      modalTitleProcedure: "¿Seguro que deseas abandonar el trámite?",
      exitBtn: "Aceptar",
      cancelBtn: "Cancelar",
    },
  },
};
