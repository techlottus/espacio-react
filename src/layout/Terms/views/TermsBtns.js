import React, { useEffect, useState } from "react";
import { Button } from "../../../components/Button/Button";
import { ProgressBar } from "../../../components/ProgressBar/ProgressBar";
import { Toggle } from "../../../components/Toggle/Toggle";
import {
  dataTerms,
  dataPrivacy,
  dataProgress,
  sendBtn,
  toggleData,
  typesTermsAndPrivacy,
} from "../../../constants/Terms.constant";

import { useDispatch, useSelector } from "react-redux";
import { useHistory } from "react-router";

import "./TermsBtns.scss";
import { patchPrivacyService } from "../../../middlewares/termsMiddelware";
import { sendInfoTM } from "../../../tagging/services/sendInfoTagManager";
import { switchOnTermsTag } from "../../../tagging/flows/termsTag";
import { testTermsTitle } from "../../../constants/test/testTerms";
const TermsBtns = ({ onClickTerms, onClickPrivacy }) => {

  const { texts } = useSelector((state) => state.texts);

  const [btnTerms, btnInitTerms] = useState({
    ...dataTerms,
    icon: "",
    disabled: true,
    title: texts?.terms.btns.btnTerms,
  });
  const [btnPrivacy, btnInitPrivacy] = useState({
    ...dataPrivacy,
    icon: "",
    title: texts?.terms.btns.btnPrivacy,
  });
  const [btnSend, setBtnSend] = useState({
    ...sendBtn,
    disabled: true,
    title: texts?.terms.btns.btnEnter,
  });
  const [isToogle, setIsToogle] = useState({
    ...toggleData
  });


  const [progressData, setProgressData] = useState({
    ...dataProgress,
    progress: 0,
    description: texts?.terms.progress.progressZero,
  });

  const [nextPrivacy, setNextPrivacy] = useState(false);
  const [nextTerms, setNextTerms] = useState(false);

  const termsStore = useSelector((state) => state.terms);

  const history = useHistory();

  const dispatch = useDispatch();

  useEffect(() => {
    window.scrollTo(0, 0);
    if (termsStore.privacy_notice_agreed) {
      btnInitPrivacy({ ...dataPrivacy, icon: "done", disabled: true });
      setProgressData({
        ...dataProgress,
        progress: 50,
        description: texts?.terms.progress.progressMiddle,
      });
      btnInitTerms({ ...dataTerms, icon: "", disabled: false });
    }
    if (termsStore.terms_and_conditions_agreed) {
      btnInitTerms({ ...dataTerms, icon: "done", disabled: true });
      setProgressData({
        ...dataProgress,
        progress: 100,
        description: texts?.terms.progress.progressComplete,
      });
    }
    if (
      termsStore.privacy_notice_agreed &&
      termsStore.terms_and_conditions_agreed
    ) {
      setBtnSend({
        ...btnSend,
        disabled: false,
      });
    }
  }, [termsStore]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    if (nextPrivacy) {
      onClickPrivacy();
    }
  }, [nextPrivacy]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    if (nextTerms) {
      onClickTerms();
    }
  }, [nextTerms]); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <div className="terms-container">
      <h2 className="termsbtns-title">{texts?.terms.title}</h2>
      <p className="termsbtns-text">{texts?.terms.subTitle}</p>

      <div className="buttons">
        <div className="button-wrap">
          <Button
            key="privacy-btn"
            data={{
              ...btnPrivacy,
              test: testTermsTitle.btnPrivacyAction
            }}
            onClick={() => setNextPrivacy((state) => !state)}
          />
        </div>
        <div className="button-wrap second-btn">
          <Button
            data={{
              ...btnTerms,
              test: testTermsTitle.btnTermsAction
            }}
            onClick={() => setNextTerms((state) => !state)}
          />
        </div>
      </div>
      <div className="newsletter-wrap">
        <Toggle
          data={{
            ...isToogle,
            test: testTermsTitle.toogleTerms
          }}
          onSwitch={(e) => {
            setIsToogle({...isToogle, status:e.detail});
            e.detail ? sendInfoTM(window, switchOnTermsTag, 'on') :  sendInfoTM(window, switchOnTermsTag, 'off')
          }}
        />
        <p className="termsbtns-text">{texts?.terms.toggle}</p>
      </div>
      <div className="progress-bar">
        <ProgressBar data={progressData} />
      </div>
      <div className="send-btn">
        <Button
          data={{
            ...btnSend,
            test: testTermsTitle.btnFinish
          }}
          onClick={() => {
            dispatch(
              patchPrivacyService(
                typesTermsAndPrivacy.promotions,
                history,
                null
              )
            );
          }}
        />
      </div>
    </div>
  );
};

export default TermsBtns;
