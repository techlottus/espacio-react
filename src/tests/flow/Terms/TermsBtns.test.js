import React from "react";
import { mount } from "enzyme";
import TermsBtns from "../../../layout/Terms/views/TermsBtns";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { act } from "react-dom/test-utils";
import { Button } from "../../../components/Button/Button";
import { Toggle } from "../../../components/Toggle/Toggle";
import { termsText } from "../../../texts/termsText";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

window.scrollTo = jest.fn();
window.dataLayer = {
  push: jest.fn()
}

let initState = {
  terms: {
    privacy_notice_agreed: false,
    terms_and_conditions_agreed: false,
  },
  texts: {
    terms: {
      ...termsText
    }
  }
};

let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <TermsBtns onClickTerms={jest.fn()} onClickPrivacy={jest.fn()} />
    </Provider>
  );
};
describe("Test <TermsBtns/> ", () => {
  beforeEach(() => {});

  test("init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
  });

  describe("should ", () => {
    test("1", () => {
      initState = {
        ...initState,
        terms: {
          ...initState.terms,
          privacy_notice_agreed: true,
        },
      };
      const wrapper = handleMount(initState);
      const progressBar = wrapper.find("ProgressBar").props();
      expect(progressBar.data.progress).toBe(50);
    });
    test("2", () => {
      initState = {
        ...initState,
        terms: {
          ...initState.terms,
          terms_and_conditions_agreed: true,
        },
      };
      const wrapper = handleMount(initState);
      const progressBar = wrapper.find("ProgressBar").props();
      expect(progressBar.data.progress).toBe(100);
    });
  });

  describe("should do next terms or privacy", () => {
    test("should do next privacy", () => {
      initState = {
        ...initState,
        terms: {
          ...initState.terms,
        },
      };
      const wrapper = handleMount(initState);
      act(() => {
        wrapper.find(Button).first().props().onClick();
      });
    });

    test("should do next Terms", () => {
      initState = {
        ...initState,
        terms: {
          ...initState.terms,
        },
      };
      const wrapper = handleMount(initState);
      act(() => {
        wrapper.find(Button).at(1).props().onClick();
      });
    });
  });

  test("should toggle button", () => {
    initState = {
      ...initState,
      terms: {
        ...initState.terms,
      },
    };
    let e = {
      detail: false,
    };
    const wrapper = handleMount(initState);
    act(() => {
      wrapper.find(Toggle).first().props().onSwitch(e.detail);
    });
  });

  test("should send terms and privacy ", () => {
    initState = {
      ...initState,
      terms: {
        ...initState.terms,
      },
    };
    const wrapper = handleMount(initState);
    act(() => {
      wrapper.find(Button).at(2).props().onClick();
    });
  });
});
