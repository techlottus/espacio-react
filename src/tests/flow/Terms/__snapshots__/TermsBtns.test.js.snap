// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <TermsBtns/>  init component 1`] = `
<div
  className="terms-container"
>
  <h2
    className="termsbtns-title"
  />
  <p
    className="termsbtns-text"
  />
  <div
    className="buttons"
  >
    <div
      className="button-wrap"
    >
      <lottus-button />
    </div>
    <div
      className="button-wrap second-btn"
    >
      <lottus-button />
    </div>
  </div>
  <div
    className="newsletter-wrap"
  >
    <lottus-toogle-switch />
    <p
      className="termsbtns-text"
    />
  </div>
  <div
    className="progress-bar"
  >
    <lottus-progress-bar />
  </div>
  <div
    className="send-btn"
  >
    <lottus-button />
  </div>
</div>
`;
